app/Http/Controllers/ApiController.php

```diff
@@ -0,0 +1,186 @@
+                    - 15&percnt; korting<br/>
+                    - 30&percnt; korting<br/>
+                    - 50&percnt; korting</td><td>";
+                $prijs_html .= implode('', array_map(function (float $disc) use ($event) {
+                    return '&euro; ' . EventPayment::calculate_price($event->prijs, $disc) . '<br/>';
+                }, Participant::INCOME_DISCOUNT_TABLE));
+                $prijs_html .= '</td>';
+            }
+
+            // Create a string with Google Maps hyperlink for the members agenda
+            $adres = $event->location->adres;
+            $plaats = $event->location->plaats;
+            if ($plaats === 'Onbekend') {
+                $kamphuis_link = $plaats;
+            } else {
+                $string = str_replace(' ', '+', $adres . ' ' . $plaats);
+                $kamphuis_link = "<a href='https://www.google.com/maps?q=" . $string . "' target='_blank'>" . $plaats . '</a>';
+            }
+
+            // Weekday table
+            $weekdays = [
+                '',
+                'Maandag',
+                'Dinsdag',
+                'Woensdag',
+                'Donderdag',
+                'Vrijdag',
+                'Zaterdag',
+                'Zondag',
+            ];
+
+            // Create data array to return
+            $data[] = [
+                'id' => $k,
+                'naam' => $naam,
+                'code' => $event->code,
+                'voordag_tekst' => ($event->type === 'kamp') ? 'Voordag:<br/>' : null,
+                'datum_voordag' => ($event->type === 'kamp') ? DateHelper::Format($event->datum_voordag) . '<br/>' : null,
+                'tijd_voordag' => ($event->type === 'kamp') ?
+                    '&nbsp;<br/>' : null,
+                'weekdag_start' => $weekdays[$event->datum_start->format('N')],
+                'datum_start' => DateHelper::Format($event->datum_start),
+                'tijd_start' => substr($event->tijd_start, 0, 5),
+                'weekdag_eind' => $weekdays[$event->datum_eind->format('N')],
+                'datum_eind' => DateHelper::Format($event->datum_eind),
+                'tijd_eind' => substr($event->tijd_eind, 0, 5),
+                'aantal_dagen' => $event->datum_eind->diffInDays($event->datum_start),
+                'kamphuis_naam' => $event->location->naam,
+                'kamphuis_adres' => $event->location->adres,
+                'kamphuis_postcode' => $event->location->postcode,
+                'kamphuis_plaats' => $event->location->plaats,
+                'kamphuis_telefoon' => $event->location->telefoon,
+                'kamphuis_website' => $event->location->website,
+                'kamphuis_mapslink' => $kamphuis_link,
+                'prijs' => $prijs_html,
+                'beschrijving' => $event->beschrijving,
+                'kleur' => $color,
+            ];
+            $k++;
+        }
+
+        return response()->json($data);
```

Comment:
Instead of doing this in the controller why not add a laravel `resource`, or a custom view model which does this. This why we separate the logic from the view logic.